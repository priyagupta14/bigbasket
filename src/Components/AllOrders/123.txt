{order.map((eachOrder) => (
            <>
              <tr className="order-value" key={eachOrder.id}>
                <td>{eachOrder.id}</td>
                <td>{new Date(eachOrder.date).toString()}</td>
              </tr>
              <Table key={eachOrder.id} productList={groupByCategory(eachOrder.items)} />
            </>
          ))}