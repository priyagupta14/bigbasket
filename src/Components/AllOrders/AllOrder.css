.orders-table-header {
  text-align: left;
}
.orders-table > thead>tr {
  display: flex;
  justify-content: space-between;
}
.orders-table{

  display: flex;

  flex-direction: column;
  justify-content: space-between;
  margin: 5rem;
}
.item-value {
  text-align: left;
}

.orders-table-header > th {
  width: 25%;
  background-color: rgba(0, 0, 0, 0.808);
  color: aliceblue;
}

.order-value > td {
  width:25%;
}
tbody>tr>td{
  width: 25%
}

.title {
  text-align: left;
}

