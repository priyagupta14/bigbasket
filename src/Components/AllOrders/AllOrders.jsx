import React, { Component } from 'react';

export default class AllOrders extends Component {
  render() {
    return (
      <div>
        <h1>Past Orders</h1>
        <hr />
      </div>
    );
  }
}
