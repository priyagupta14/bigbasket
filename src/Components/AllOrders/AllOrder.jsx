/* eslint-disable react/prop-types */
import React from 'react';
import './AllOrder.css';
import Table from '../Table/Table';
import groupByCategory from '../../Utils/GroupBy/groupByCategory';

const AllOrders = ({ order }) => {
  const total = order.map((eachOrder) => (
    eachOrder.items.reduce((acc, curVal) => acc + curVal.price, 0)));
  return (
    <div>
      <h1>Past Orders</h1>
      <hr />
      <table className="orders-table" border="1">
        <thead>
          <tr className="orders-table-header">
            <th>Order</th>
            <th>items</th>
            <th>Order Date</th>
            <th>Amount</th>
          </tr>
        </thead>

        {order.map((eachOrder, index) => (
          <React.Fragment key={eachOrder.id}>
            <tbody>
              <tr className="order-value">
                <td>{eachOrder.id}</td>
                <td>{eachOrder.items.length}</td>
                <td>{new Date(eachOrder.date).toString()}</td>
                <td>{total[index]}</td>
              </tr>
            </tbody>
            <Table
              key={eachOrder.id + eachOrder
                .items.length}
              productList={groupByCategory(eachOrder.items)}
            />
          </React.Fragment>

        ))}
      </table>
    </div>
  );
};
export default AllOrders;
