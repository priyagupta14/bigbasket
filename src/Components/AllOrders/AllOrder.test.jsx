import React from 'react';
import {
  fireEvent, render, screen,
} from '@testing-library/react';
import { Route, BrowserRouter } from 'react-router-dom';
import AllOrders from './AllOrder';

describe(AllOrders.name, () => {
  const mockProps = {
    order: [
      {
        items: [
          {
            id: 1,
            name: 'pp',
            count: 20,
            price: 12,
            category: 'idk',
          },
        ],
        id: 1,
        date: 123,
      },
    ],
    productList: [],
  };
  it('should match the snapShot', () => {
    const { container } = render(<BrowserRouter><AllOrders {...mockProps} /></BrowserRouter>);
    expect(container).toMatchSnapshot();
  });
  it('should display Past Orders and headers', () => {
    render(<AllOrders {...mockProps} />);
    screen.getByText('Past Orders');
    screen.getByText('Order');
    screen.getByText('items');
    screen.getByText('Order Date');
    screen.getByText('Amount');
  });
});
