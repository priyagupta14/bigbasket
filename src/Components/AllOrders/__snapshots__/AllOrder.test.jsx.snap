// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllOrders should match the snapShot 1`] = `
<div>
  <div>
    <h1>
      Past Orders
    </h1>
    <hr />
    <table
      border="1"
      class="orders-table"
    >
      <thead>
        <tr
          class="orders-table-header"
        >
          <th>
            Order
          </th>
          <th>
            items
          </th>
          <th>
            Order Date
          </th>
          <th>
            Amount
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class="order-value"
        >
          <td>
            1
          </td>
          <td>
            1
          </td>
          <td>
            Thu Jan 01 1970 05:30:00 GMT+0530 (India Standard Time)
          </td>
          <td>
            12
          </td>
        </tr>
      </tbody>
      <table
        border="1"
        class="cart-table"
      >
        <thead>
          <tr
            class="cart-table-header"
          >
            <th>
              ITEM DESCRIPTION
            </th>
            <th>
              UNIT PRICE
            </th>
            <th>
              QUANTITY
            </th>
            <th>
              SUBTOTAL
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            class="title"
          >
            <td>
              idk
            </td>
          </tr>
          <tr
            class="item-value"
          >
            <td>
              pp
            </td>
            <td>
              12
            </td>
            <td>
              20
            </td>
            <td>
              240
            </td>
          </tr>
        </tbody>
      </table>
    </table>
  </div>
</div>
`;
